guess-the-country/country-backend/country/model/predict.py

```diff
@@ -44,9 +46,7 @@ def preprocess(img, IMG_SIZE=224):
     img_rgb = cv2.cvtColor(img, cv2.COLOR_BGR2RGB)
     # resize image to match model's expected sizing
     img_resize = cv2.resize(img_rgb, (IMG_SIZE, IMG_SIZE))
-    image = tf.keras.applications.mobilenet_v2.preprocess_input(img_resize)
-    pre_image = image.reshape(-1, IMG_SIZE, IMG_SIZE, 3)
-    return pre_image
```

Comment:
Das verstehe ich nicht. Wo findet denn jetzt bei einer "normalen" Vorhersage das Preprocessing statt?

Reply:
... das hatte ich nicht auf dem Schirm gehabt :/

Reply:
Ist es vertretbar preprocess so zu lassen wie es im master ist und dazu eine preprocess_visualime zu erstellen, welche die ersten 2 Befehle von preprocess beinhaltet? Diese würde ich dann in explain.py.explain verwenden.

Reply:
Ja, du kannst die Funktion sogar auch "preprocess" nennen. Ich würde das Preprocessing hier genau so lassen, wie es für die Predictions sein muss und dann ein zweites Preprocessing für die Explainer schreiben, das genau für diesen Zweck angepasst ist.